File: app/src/main/java/com/sum/hi/ui/demo/framework/mvp/MainActivity.java
package com.sum.hi.ui.demo.framework.mvp;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @创建者 mingyan.su
 * @创建时间 2022/07/24 11:42
 * @类描述 ${TODO}
 */
class MainActivity extends AppCompatActivity implements HomeContract.View {

    private HomePresenter presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        presenter = new HomePresenter();
        //绑定view的关系
        presenter.attach(this);
        presenter.getUserInfo();
    }

    @Override
    public void onGetUserInfo(Object data, String errorMsg) {

    }

    @Override
    public void onSaveUserInfo(Object data, String errorMsg) {

    }

    @Override
    public boolean isAlive() {
        return !isDestroyed() && !isFinishing();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //移除view
        presenter.detach();
    }
}

File: hi_hilt/src/main/java/com/sum/hi/hi_hilt/MainActivity.kt
package com.sum.hi.hi_hilt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

//标记@AndroidEntryPoint
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    //标记@Inject
    @Inject
    lateinit var iLoginService: ILoginService
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /**
         * 思考：1.ILoginService是如何被创建，如何被赋值的？
         *      2.LoginServiceImpl(context) 是如何被传递进去的？
         *      3.全局单例是如何实现的？
         *      把这三个掌握了，Hilt的运行时原理也就掌握了
         */
        iLoginService.login()
    }
}